### ISSUE TITLE
Unable to place cursor via mouse or arrow keys to right of a line terminating inline void

### ISSUE THREAD

**Issue by instadrew** [ianstormtaylor/slate#4839]
**Description**
Unable to place cursor via mouse or arrow keys to right of a line terminating inline void.

**Recording**
Can't really provide a useful video because the issue is the absence of a behavior, not the presence of one.

**Sandbox**
https://codesandbox.io/s/slate-js-0-72-8-custom-inline-void-element-q5exj?file=/src/App.js

**Steps**
To reproduce the behavior:
1. Go to https://codesandbox.io/s/slate-js-0-72-8-custom-inline-void-element-q5exj?file=/src/App.js
2. Note that what appears in the editor as "inline_void_text" with a grey background are all inline voids.
3. In the Slate editor demo area, on either line of text, click to the right of the second "inline_void_text" at the end of the line.  You should see that no cursor is displayed to the right of "inline_void_text".
4. On the same line, place the cursor a little to the left of the same "inline_void_text" and try to right arrow key to the right of it.  You should see that on the first line, the cursor jumps from the left of the "inline_void_text" to the beginning of the second line without ever appearing to the right of "inline_void_text".  On the second line, however, the cursor stops at the left of "inline_void_text".
5. The first "inline_void_text" on both the first and second lines both allow the cursor to be placed both to the left and to the right of themselves, and allow the arrow keys to move the cursor from one side to the other.

**Expectation**
I'd expect to be able to place the cursor via mouse or arrow keys to the right of an inline void wherever it appears because I might want to add text to the right of it.

**Environment**
- Slate Version: 0.72.8
- Operating System: MacOS 12.2.1 (Monterey)
- Browser: Chrome Version 98.0.4758.80 (Official Build) (x86_64)

**Comment by zbeyens** [comment]
I got the same issue, clicking after the void node will select the void node instead of the next empty text node.

**Comment by zbeyens** [comment]
I can't reproduce it with slate 0.71.0

**Comment by instadrew** [comment]
@zbeyens Thanks....I believe I have `{...attributes}` on the top element of the rendering leaf in the codesandbox example I provided, but the issue still happens there.  Is the below what you meant?

```
const Leaf = ({ attributes, children, leaf }) => {
  if (leaf.bold) {
    children = <strong>{children}</strong>;
  }

  if (leaf.code) {
    children = <code>{children}</code>;
  }

  if (leaf.italic) {
    children = <em>{children}</em>;
  }

  if (leaf.underline) {
    children = <u>{children}</u>;
  }

  return <span {...attributes}>{children}</span>;
};
```

**Comment by cameronbraid** [comment]
Another possibly related issue is demonstrated in you example if you select first inline void element then right arrow the cursor jumps one character trailing the inline void and you have to left arrow to go to the correct place.

**Comment by instadrew** [comment]
@cameronbraid Yes, I agree.  The default behavior of inline voids does not seem fully worked out to me.

**Comment by dylans** [comment]
In general I always add a zero-width character at the end of the line when adding a void. Even then, I believe that the change in https://github.com/ianstormtaylor/slate/pull/4952 causes a further regression (pushes the cursor to the next line).
